packages/cli/src/evaluation/test-runner/test-runner.service.ee.ts

```diff
@@ -180,46 +180,58 @@ export class TestRunnerService {
 
 		// 2. Run over all the test cases
 
-		await this.testRunRepository.markAsRunning(testRun.id);
+		await this.testRunRepository.markAsRunning(testRun.id, pastExecutions.length);
 
 		// Object to collect the results of the evaluation workflow executions
 		const metrics = new EvaluationMetrics(testMetricNames);
 
 		for (const { id: pastExecutionId } of pastExecutions) {
-			// Fetch past execution with data
-			const pastExecution = await this.executionRepository.findOne({
-				where: { id: pastExecutionId },
-				relations: ['executionData', 'metadata'],
-			});
-			assert(pastExecution, 'Execution not found');
-
-			const executionData = parse(pastExecution.executionData.data) as IRunExecutionData;
-
-			// Run the test case and wait for it to finish
-			const testCaseExecution = await this.runTestCase(workflow, executionData, user.id);
-
-			// In case of a permission check issue, the test case execution will be undefined.
-			// Skip them and continue with the next test case
-			if (!testCaseExecution) {
-				continue;
+			try {
+				// Fetch past execution with data
+				const pastExecution = await this.executionRepository.findOne({
+					where: { id: pastExecutionId },
+					relations: ['executionData', 'metadata'],
+				});
+				assert(pastExecution, 'Execution not found');
+
+				const executionData = parse(pastExecution.executionData.data) as IRunExecutionData;
+
+				// Run the test case and wait for it to finish
+				const testCaseExecution = await this.runTestCase(workflow, executionData, user.id);
+
+				// In case of a permission check issue, the test case execution will be undefined.
+				// Skip them, increment the failed count and continue with the next test case
+				if (!testCaseExecution) {
+					await this.testRunRepository.incrementFailed(testRun.id);
+					continue;
+				}
+
+				// Collect the results of the test case execution
+				const testCaseRunData = testCaseExecution.data.resultData.runData;
+
+				// Get the original runData from the test case execution data
+				const originalRunData = executionData.resultData.runData;
+
+				// Run the evaluation workflow with the original and new run data
+				const evalExecution = await this.runTestCaseEvaluation(
+					evaluationWorkflow,
+					originalRunData,
+					testCaseRunData,
+				);
+				assert(evalExecution);
+
+				// Extract the output of the last node executed in the evaluation workflow
+				metrics.addResults(this.extractEvaluationResult(evalExecution));
+
+				if (evalExecution.data.resultData.error) {
+					await this.testRunRepository.incrementFailed(testRun.id);
+				} else {
+					await this.testRunRepository.incrementPassed(testRun.id);
+				}
+			} catch (e) {
+				// In case of an unexpected error, increment the failed count and continue with the next test case
```

Comment:
Is this error a result of the user or developer? Should we handle it somehow? E.g. capture it (to sentry) or at least log it? Worried it might hide possible bugs

Reply:
There should not be any "expected" errors here, added reporting to Sentry